### PR TITLE
Update cli.eval to have the same YAML format as other cli files

### PR DESCRIPTION
**Pull request by gorold**
1. cli/conf/eval/default.yaml now follows a similar format to pretrain and finetune
2. move `patch_size` and `context_length` parameters to the model specification to generalise the script - patch_size was a model specific parameter, move those to the model config file

### CODE REVIEW

File: cli/eval.py [hunk 1]
```diff
@@ -32,14 +30,11 @@ def main(cfg: DictConfig):
     test_data, metadata = call(cfg.data)
     batch_size = cfg.batch_size
     while True:
-        model = call(cfg.load_from_checkpoint)(
+        model = call(cfg.model, _partial_=True)(
```
Comment (liu-jc) [inline cli/eval.py:33]: I am curious why it should be `_partial_` here, although I found the previous corresponding part is also `_partial_: true`

File: cli/conf/eval/default.yaml
Comment (liu-jc) [inline cli/conf/eval/default.yaml:1]: LGTM. But my concern is that if we try to update this cli.eval to be the same format as others, the readme needs to be changed accordingly? For example, `patch_size` -> `model.patch_size`.
Therefore, I wonder whether it's necessary to have this change. The advantage of this change to me is that the loaded model can be more flexible as defined in different yaml files of the `conf/eval/model` folder.

**Review by liu-jc — COMMENTED** [review]
This pull request basically increases the flexiblity of defining which models to load for evaluation. If we want to have this change, we may have to change the readme again for the evaluation part. I am not sure whether it will increase the obstacles for users to play with our code.

File: cli/eval.py (hunk 1 above)
Comment (gorold) [reply cli/eval.py:33]: It's a partial instantiation, similar to functools.partial - check out the [hydra docs](https://hydra.cc/docs/advanced/instantiate_objects/overview/#partial-instantiation)

File: cli/conf/eval/default.yaml
Comment (gorold) [reply cli/conf/eval/default.yaml:1]: Thanks, will make the changes to the readme

File: pyproject.toml [hunk 2]
```diff
@@ -8,7 +8,7 @@ dynamic = [
   "version",
 ]
 dependencies = [
-  "torch>=2.0",
+  "torch>=2.1",
```
Comment (liu-jc) [inline pyproject.toml:11]: Any reason why we are also changing the dependencies?

File: README.md [hunk 3]
```diff
@@ -192,12 +196,13 @@ echo "LSF_PATH=PATH_TO_TSLIB/dataset" >> .env
 Thereafter, simply run the following script with the predefined [Hydra config file](cli/conf/eval/data/lsf_test.yaml):
 ```shell
 python -m cli.eval \ 
+  run_name=example_eval \
```
Comment (liu-jc) [inline README.md:199]: Should we use different run_name for two different examples? Otherwise the results would be placed under the same folder? Maybe change this run name to `example_lsf_eval`?

Comment (gorold) [reply README.md]: done

**Review by liu-jc — APPROVED** [review]
